src/ziggurat/streams.clj

```diff
@@ -92,6 +97,22 @@
         additional-tags  {:topic_name topic-entity-name}]
     (.transform stream-builder (transformer-supplier metric-namespace oldest-processed-message-in-s additional-tags) (into-array [(.name (store-supplier-builder))]))))
 
+(defn- traced-handler-fn [handler-fn channels message-payload topic-entity]
+  (let [parent-ctx (TracingKafkaUtils/extractSpanContext (:headers message-payload) tracer)
+        span (-> tracer
+                 (.buildSpan "Message-Handler")
+                 (.asChildOf parent-ctx)
+                 (.withTag (.getKey Tags/SPAN_KIND) Tags/SPAN_KIND_CONSUMER)
+                 (.withTag (.getKey Tags/COMPONENT) "lambda")
+                 (.start))]
+    (try
+      (.activate (.scopeManager tracer) span)
+      ((mapper-func handler-fn channels) (->MessagePayload (:value message-payload) topic-entity (:headers message-payload)))
+      (catch Exception e
+        (log/error "Exception while executing handler function " e))
```

Comment:
why is there a generic exception catch here? This will lead to skipping the message and data loss right?

Reply:
oh ya. this was added for testing the error scenario. will remove the catch